internal/controller/status/status.go

```diff
@@ -0,0 +1,86 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	danaiov1alpha1 "github.com/dana-team/nfspvc-operator/api/v1alpha1"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	storagePhaseUnknown  = "Unknown"
+	storagePhaseNotFound = "NotFound"
+)
+
+// UpdateNfsPvcStatus fetches the phase of the pv and the pvc that is created by the nfspvc and updates the nfspvc status.
+func UpdateNfsPvcStatus(ctx context.Context, nfspvc danaiov1alpha1.NfsPvc, log logr.Logger, k8sClient client.Client) error {
+	nfspvcObject := danaiov1alpha1.NfsPvc{}
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: nfspvc.Namespace, Name: nfspvc.Name}, &nfspvcObject); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info(fmt.Sprintf("Didn't find NfsPvc: %s, from the namespace: %s", nfspvc.Name, nfspvc.Namespace))
+			return nil
+		}
+		return fmt.Errorf("failed to get NfsPvc: %s", err.Error())
+	}
+
+	pvcPhase := getPVCStatus(ctx, nfspvc, k8sClient)
+	pvPhase := getPVStatus(ctx, nfspvc, k8sClient)
+
+	if pvcPhase != nfspvc.Status.PvcPhase || pvPhase != nfspvc.Status.PvPhase {
+		return updateStatus(ctx, pvcPhase, pvPhase, nfspvcObject, k8sClient)
+	}
+	return nil
+}
```

Comment:
It's the `status` package and this is all code for `NfsPvc`, so this can just be renamed as `Update`. Also why are we doing a GET here? This is already handled in the function it calls to actually perform the update.

Reply:
I think this is to handle conflicts(If the object is changed in between)

Reply:
Right but the conflict handling is already implemented in the `updateStatus` code